tensorpipe/test/channel/channel_test_cuda.cc

```diff
@@ -236,6 +236,236 @@ class SendAcrossDevicesTest : public ClientServerChannelTestCase<CudaBuffer> {
 
 CHANNEL_TEST(CudaChannelTestSuite, SendAcrossDevices);
 
+class SendReverseAcrossDevicesTest
```

Comment:
these two tests are mostly duplicated from the above test, and the only differences are src/dst devices.